core/src/main/scala/org/apache/spark/sql/hive/SnappyStoreHiveCatalog.scala

```diff
@@ -297,7 +297,18 @@ class SnappyStoreHiveCatalog(externalCatalog: SnappyExternalCatalog,
             val builder = new MetadataBuilder
             builder.withMetadata(f.metadata).putString("name", name).build()
           } else {
-            f.metadata
+            f.dataType match {
+              case s: StringType =>
+                if (!f.metadata.contains(Constant.CHAR_TYPE_BASE_PROP)) {
+                  val builder = new MetadataBuilder
+                  builder.withMetadata(f.metadata).putString(Constant.CHAR_TYPE_BASE_PROP,
+                    "STRING").putLong(Constant.CHAR_TYPE_SIZE_PROP, Constant.MAX_VARCHAR_SIZE)
```

Comment:
What does the combination of "STRING" and MAX_VARCHAR_SIZE mean? Shouldn't there be no size specification if "STRING" is the type?